physx.js

```diff
@@ -704,6 +704,94 @@ const physxWorker = (() => {
       throw new Error('raycastPhysicsArray error');
     }
   };
+
+  w.doCut = (
+    positions,
+    numPositions,
+    normals,
+    numNormals,
+    uvs,
+    numUvs,
+    faces,
+    numFaces,
+
+    position,
+    quaternion,
+    scale,
+  ) => {
+    const allocator = new Allocator();
+
+    const positionsTypedArray = allocator.alloc(Float32Array, numPositions);
+    positionsTypedArray.set(positions);
+
+    const normalsTypedArray = allocator.alloc(Float32Array, numNormals);
+    normalsTypedArray.set(normals);
+
+    const uvsTypedArray = allocator.alloc(Float32Array, numUvs);
+    uvsTypedArray.set(uvs);
+
+    const facesTypedArray = allocator.alloc(Uint32Array, numFaces)
+    facesTypedArray.set(faces);
+
+    const positionTypedArray = allocator.alloc(Float32Array, 3);
+    positionTypedArray.set(position);
+
+    const quaternionTypedArray = allocator.alloc(Float32Array, 4);
+    quaternionTypedArray.set(quaternion);
+
+    const scaleTypedArray = allocator.alloc(Float32Array, 3)
+    scaleTypedArray.set(scale);
+
+    const numOutPositionsTypedArray = allocator.alloc(Uint32Array, 2);
+    const numOutNormalsTypedArray = allocator.alloc(Uint32Array, 2);
+    const numOutUvsTypedArray = allocator.alloc(Uint32Array, 2);
+    const numOutFacesTypedArray = allocator.alloc(Uint32Array, 2);
+
+    // output lenght may bigger than input length?
+    // todo: how to pass unknonwn length output from wasm to js?
+    // see https://rob-blackbourn.github.io/blog/webassembly/wasm/array/arrays/javascript/c/2020/06/07/wasm-arrays.html #Passing output arrays to wasm
+    // whether this tut has solution?
+    const outPositionsTypedArray = allocator.alloc(Float32Array, numPositions * 10); // todo: do not hard code alloc length.
+    const outNormalsTypedArray = allocator.alloc(Float32Array, numNormals * 10); // todo: do not hard code alloc length.
+    const outUvsTypedArray = allocator.alloc(Float32Array, numUvs * 10); // todo: do not hard code alloc length.
+    const outFacesTypedArray = allocator.alloc(Uint32Array, numFaces * 10); // todo: do not hard code alloc length.

```

Comment:
This is good use of the `Allocator` but we need to `free()` it when we are done or we leak memory from C++.

Reply:
https://github.com/webaverse/app/blob/aa2905e229fd04695742b5de8534c07e1f707571/physx.js#L871

Reply:
Thanks! Added https://github.com/webaverse/app/pull/2211/commits/c011b3e4757c1e2119d776146a5aba8e57d41d7c .